holidays/financial/ny_stock_exchange.py

```diff
@@ -43,53 +30,43 @@ class NewYorkStockExchange(HolidayBase, ChristianHolidays, InternationalHolidays
     """
 
     market = "NYSE"
+    observed_label = "%s (Observed)"
 
     def __init__(self, *args, **kwargs):
         ChristianHolidays.__init__(self)
         InternationalHolidays.__init__(self)
         StaticHolidays.__init__(self, NewYorkStockExchangeStaticHolidays)
+        kwargs.setdefault("observed_rule", SAT_TO_PREV_FRI + SUN_TO_NEXT_MON)
         super().__init__(*args, **kwargs)
 
-    def _add_observed_holiday(self, name: str, dt: date) -> None:
-        if self._is_saturday(dt):
-            dt += td(days=-1)
-            name = f"{name} (Observed)"
-        elif self._is_sunday(dt):
-            dt += td(days=+1)
-            name = f"{name} (Observed)"
-
-        self._add_holiday(name, dt)
+    def _add_observed_holiday(self, name, *args) -> None:
+        dt = date(self._year, *args)
+        is_obs, _ = self._add_observed(dt, name)
+        if not is_obs:
+            self._add_holiday(name, dt)
```

Comment:
It didn't seem right to have two similar `_add_observed...` method. Could you check if this suggestion covers your needs:
```suggestion
    def _add_observed(self, name, *dt) -> None:  # type: ignore[override]
        dt = dt if len(dt) > 1 else dt[0]
        is_observed, _ = super()._add_observed(dt, name)  # type: ignore[arg-type, assignment]
        if not is_observed:
            self._add_holiday(name, dt)  # type: ignore[assignment]

```

You'll need to replace all `_add_observed_holiday` cases to `_add_observed`. Please note the args order: name, then date args

Reply:
We have `_move_holiday()` for such cases, I forgot about it. :)

Reply:
That's a lot better!